http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -138,13 +138,21 @@ class LeoPubsubMessageSubscriber[F[_]](
   private[monitor] def messageHandler(event: Event[LeoPubsubMessage]): F[Unit] = {
     val traceId = event.traceId.getOrElse(TraceId("None"))
     val now = Instant.ofEpochMilli(com.google.protobuf.util.Timestamps.toMillis(event.publishedTime))
-    implicit val appContext = Ask.const[F, AppContext](AppContext(traceId, now))
+    implicit val ev = Ask.const[F, AppContext](AppContext(traceId, now, span = None))
+    childSpan(event.msg.messageType.asString).use { implicit ev =>
```

Comment:
Generate a root `Span` at the pubsub message handler entry point, and pass it downstream in the `AppContext`.